### PR TITLE
[yasm] Fix feature tools

### PR DESCRIPTION
**Pull request by JackBoosY**
When building dynamic yasm, its tools need to dynamically call the library `yasmstd`, so manually copy the library to tools.
Add support with arm / linux /osx.

Related: https://github.com/microsoft/vcpkg/pull/18816.

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

File: ports/yasm/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "yasm",
+  "version-semver": "1.3.0",
+  "port-version": 1,
+  "description": "Yasm is a complete rewrite of the NASM assembler under the “new” BSD License.",
```
Comment (ras0219-msft) [inline ports/yasm/vcpkg.json:5]:
```suggestion
  "description": "Yasm is a complete rewrite of the NASM assembler under the new BSD License.",
```

Let's get rid of the excessive unicode while we're here.

File: ports/yasm/fix-arm-build.patch [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
++++ b/cmake/modules/YasmMacros.cmake
+@@ -58,7 +58,10 @@ macro (YASM_ADD_MODULE _module_NAME)
+ endmacro (YASM_ADD_MODULE)
+ 
+ macro (YASM_GENPERF _in_NAME _out_NAME)
+-    get_target_property(_tmp_GENPERF_EXE genperf LOCATION)
++    find_program(_tmp_GENPERF_EXE genperf)
++    if (NOT _tmp_GENPERF_EXE)
++        get_target_property(_tmp_GENPERF_EXE genperf LOCATION)
++    endif()
```
Comment (ras0219-msft) [inline ports/yasm/fix-arm-build.patch:13]: Why is this change needed? This looks like it is attempting to support cross compilation of yasm, which is not a use case for vcpkg.

Specifically, `yasm:arm64-windows --host-triplet=x64-windows` is expected to fail. `yasm:arm64-windows --host-triplet=arm64-windows` is expected to succeed, on an arm64-based system.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/yasm/fix-arm-build.patch (hunk 2 above)
Comment (BillyONeal) [reply ports/yasm/fix-arm-build.patch]: >Why is this change needed? This looks like it is attempting to support cross compilation of yasm, which is not a use case for vcpkg.

This seems a reasonable use case for vcpkg. Yes, the tools that the port produces aren't useful as far as vcpkg cares given that we don't want to be a tool distribution platform. However, yasm also produces a library folks can link to to, presumably, assemble things in their own programs if they want.

I think there are 2 possible outcomes:

* If we want to provide that library, then Jack's change is correct in spirit, in that it ensures that library is available even for non-host architectures. (I might quibble in asking if there is some way to do it without patching but that's really a nitpick)
* If we only care about the tools here, then indeed the port should be made `"supports": "native"` and the patches removed since we will never build it in a cross scenario.

After the above we should point yasm-tool and yasm-tool-helper consumers at this port with deprecation messages since that whole infrastructure is a workaround from the before tool ports times.

Comment (ras0219-msft) [reply ports/yasm/fix-arm-build.patch]: If `yasm` is providing a library, then the tool should probably be moved to a feature.

Comment (JackBoosY) [reply ports/yasm/fix-arm-build.patch]: @ras0219-msft Resolved.

**Review by BillyONeal — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
I didn't get a reply from Robert but this didn't seem to be an over-his-dead-body thing so I just merged this
